File: 724_find_pivot/FindPivot.java
public class FindPivot {
    public int pivotIndex(int[] nums) {
        int presum = 0;
        for (int element : nums) {
            presum += element;
        }
        int leftSum = 0;
        for (int i = 0; i < nums.length; i++) {
            if (2 * leftSum == presum - nums[i]) {
                return i;
            }
            leftSum += nums[i];
        }
        return -1;
    }

    public static void main(String[] args) {
        int[] nums = { 1, 7, 3, 6, 5, 6 };
        FindPivot fp = new FindPivot();
        int pivotIndex = fp.pivotIndex(nums);
        System.out.println(pivotIndex);
    }
}

File: 724_find_pivot/findPivot.js
let pivotIndex = (nums) => {
  let preSum = 0;
  let leftSum = 0;
  nums.forEach((element) => {
    preSum += element;
  });
  for (let i = 0; i < nums.length; i++) {
    const element = nums[i];
    if (2 * leftSum === preSum - nums[i]) {
      return i;
    }
    leftSum += nums[i];
  }
  return -1;
};

let nums = [1, 7, 3, 6, 5, 6];

console.log(pivotIndex(nums));
